src/librustc_codegen_llvm/debuginfo/metadata.rs

```diff
@@ -1366,7 +1367,7 @@ impl EnumMemberDescriptionFactory<'ll, 'tcx> {
                             let value = (i.as_u32() as u128)
                                 .wrapping_sub(niche_variants.start().as_u32() as u128)
                                 .wrapping_add(niche_start);
-                            let value = value & ((1u128 << niche.value.size(cx).bits()) - 1);
+                            let value = truncate(value, niche.value.size(cx));
```

Comment:
Since you're touching this code, can you add the `assert_eq!(value as u64 as u128, value);` I'm suggesting in https://github.com/rust-lang/rust/pull/55701#discussion_r270227197?